frontend/src/features/public-form/components/FormFields/FormFields.tsx

```diff
@@ -31,18 +33,44 @@ export const FormFields = ({
   colorTheme,
   onSubmit,
 }: FormFieldsProps): JSX.Element => {
+  const [searchParams] = useSearchParams()
+
+  const fieldPrefillMap = useMemo(() => {
+    const queryParams = Object.fromEntries([...searchParams])
+    const ids = new Set(Object.keys(queryParams))
+
+    // Return object containing field id and query param value only if id exists in form fields.
+    return formFields.reduce((acc, field) => {
+      if (
+        field.fieldType === BasicField.ShortText &&
+        field.allowPrefill &&
+        ids.has(field._id)
+      ) {
+        acc[field._id] = queryParams[field._id]
+      }
+      return acc
+    }, {} as Record<string, string>)
```

Comment:
There's no need to parse the full searchParams and make local structures for them, [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) already has the APIs to do the job:
```suggestion
    // Return object containing field id and query param value only if id exists in form fields.
    return formFields.reduce((acc, field) => {
      if (
        field.fieldType === BasicField.ShortText &&
        field.allowPrefill &&
        searchParams.has(field._id)
      ) {
        acc[field._id] = searchParams.get(field._id)
      }
      return acc
    }, {} as Record<string, string>)
```

Reply:
resolved in 5c33a5a